### PR TITLE
Add installation as snap instructions

### PR DESCRIPTION
**Pull request by jpm-canonical**
Refactor Chip Tool docs to include installation from a snap. This allows usage of the tool on common OSs like Ubuntu without having to clone this repo and compile the tool from source.

The snap package is hosted on the [snap store](https://snapcraft.io/chip-tool). This listing indicates that the maintainer of the snap is Canonical - a [CSA member](https://csa-iot.org/member/canonical/). The listing also links to the [source code](https://github.com/canonical/chip-tool-snap) of the snap package on Github.

### CODE REVIEW

**Comment by andy31415** [comment]
@jpm-canonical please note `Make sure you delete these instructions (to prove you have read them).` in the summary and delete the section.

File: docs/development_controllers/chip-tool/chip_tool_guide.md [hunk 1]
```diff
@@ -9,33 +9,27 @@ the setup payload or performing discovery actions.
...
-> `/tmp/chip_tool_config.ini` file. Deleting this and other `.ini` files in the
-> `/tmp` directory can sometimes resolve issues related to stale configuration.
+On devices running SnapD, like Ubuntu, the CHIP Tool can be installed from the
+Snap Store. To do this, run:
 
-> **Note:** To make the configuration persistent (since `/tmp` directory might
-> be flushed at each reboot) you can change the directory where CHIP Tool caches
-> its configuration by using the option `--storage-directory`
+```
+sudo snap install chip-tool
```
Comment (andy31415) [inline docs/development_controllers/chip-tool/chip_tool_guide.md:18]: who maintains this snap build? How can people ensure its authenticity and safety?

Comment (andy31415) [reply docs/development_controllers/chip-tool/chip_tool_guide.md:18]: Link to the respository for issue reporting and tracking may also be useful

**Review by farshidtz — CHANGES_REQUESTED** [review]

**Comment by jpm-canonical** [comment]
> @jpm-canonical please note `Make sure you delete these instructions (to prove you have read them).` in the summary and delete the section.

Yes, sorry. This was supposed to be a draft PR until we have internally reviewed it, as which time the description would be fixed too. I'll ping you when we are happy and it's ready for upstream review.

File: docs/development_controllers/chip-tool/chip_tool_guide.md (hunk 1 above)
Comment (jpm-canonical) [reply docs/development_controllers/chip-tool/chip_tool_guide.md:20]: We've added a link to the [snap store listing](https://snapcraft.io/chip-tool). This listing indicates that the maintainer of the snap is Canonical - a [CSA member](https://csa-iot.org/member/canonical/). The listing also links to the [source code](https://github.com/canonical/chip-tool-snap) of the snap package on Github.

**Review by farshidtz — APPROVED** [review]
LGTM

**Review by woody-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**